### PR TITLE
docs: add migration guide from 0.12 to bonsai

### PR DESCRIPTION
**Pull request by Walther**
This adds a rudimentary migration guide from 0.12 to Bonsai.

This is a bit of a minimum viable tutorial for pre-release purposes only. We should replace this with better documentation and a blog post later on.

Squash before merging to clean up the history.

### CODE REVIEW

File: docs/tutorials/migrating-to-bonsai.md [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+- `dev-mode` has been renamed to `sync`, both in the configuration as well as on the CLI
+- `garden delete` has been renamed to `garden cleanup`
+- `garden delete env` has been renamed to `garden cleanup namespace` with an alias of `garden cleanup ns`
+- `dotIgnoreFiles` has been renamed to `dotIgnoreFile` and only supports one file
+- removed default `environments`, please specify the field in project configuration
+- template configurations now use `camelCase` everywhere, no more `snake_case` or `kebab-case`
+- the deprecated `hot-reload` has been removed, use `sync` instead
+- the deprecated `cluster-docker` build mode has been removed, use `cluster-buildkit` or `kaniko` instead
+- dropped support for deploying an in-cluster registry, see the [in-cluster build documentation](../k8s-plugins/advanced/in-cluster-building.md)
+- dropped support for Kubernetes version `1.14` and below
```
Comment (vvagaytsev) [inline docs/tutorials/migrating-to-bonsai.md:23]: Haven't we dropped `1.18` and below? Officially we support the latest 6 versions. Our integration tests check the versions `1.19 -> 1.26` (so the latest 8 versions, maybe we should also drop 1.19 and 1.20?)

Comment (Walther) [reply docs/tutorials/migrating-to-bonsai.md:23]: Good question! I pulled this from the Breaking Changes plan/wishlist on Notion, which mentioned 1.14 specifically.
I'll remove this line and we can clarify further for the proper release - this probably doesn't matter much for the pre-release (and pre-release only version of this doc)

File: docs/tutorials/migrating-to-bonsai.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+- `cert-manager` integration has been deprecated. New documentation coming soon here: [ext dns and cert manager example](https://github.com/garden-io/garden/pull/3988)
+- `dev-mode` has been renamed to `sync`, both in the configuration as well as on the CLI
+- `garden delete` has been renamed to `garden cleanup`
+- `garden delete env` has been renamed to `garden cleanup namespace` with an alias of `garden cleanup ns`
+- `dotIgnoreFiles` has been renamed to `dotIgnoreFile` and only supports one file
+- removed default `environments`, please specify the field in project configuration
+- template configurations will use `camelCase` everywhere, no more `snake_case` or `kebab-case`: [tracking issue](https://github.com/garden-io/garden/issues/3513)
+- the deprecated `hot-reload` has been removed, use `sync` instead
+- the deprecated `cluster-docker` build mode has been removed, use `cluster-buildkit` or `kaniko` instead
+- dropped support for deploying an in-cluster registry, see the [in-cluster build documentation](../k8s-plugins/advanced/in-cluster-building.md)
```
Comment (Orzelius) [inline docs/tutorials/migrating-to-bonsai.md:22]: I think the link should be to the remote registry configuration guide instead
```suggestion
ropped support for deploying an in-cluster registry, see the [in-cluster build documentation](../k8s-plugins/remote-k8s/configure-registry/README.md)
```

**Review by Orzelius — COMMENTED** [review]
I think the following is worth mentioning:
* Where is the reference documentation for actions
* Modules can't depend on actions
* Modules can't reference actions
* Actions can reference and depend on modules, by referencing the actions that are generated from modules
* Converting your project to the actions config can be performed gradually by starting from the end of the dependency tree

I think adding this resolution flow might make it more understandable:
1. modules are resolved
2. modules are converted to actions
3. actions are resolved

**Comment by vvagaytsev** [comment]
Should we also include #3512 in the breaking changes list?

File: docs/tutorials/migrating-to-bonsai.md (hunk 2 above)
Comment (vvagaytsev) [reply docs/tutorials/migrating-to-bonsai.md:22]: "d" is missing in the suggestion :)

Comment (Orzelius) [reply docs/tutorials/migrating-to-bonsai.md:22]: thanks, fixed the suggestion

**Review by vvagaytsev — APPROVED** [review]
Great job, thanks! 🚀

**Pull request merged**